### PR TITLE
Add a prow job to validate `metadata.yaml` files

### PR DESCRIPTION
**Pull request by A-Hilaly**
This patch introduces a new prow job that validates the `metadata.yaml`
files in service controller repositories. It does so by making sure that
the fields are not empty and making sure that the provided URLs are
working properly.

Signed-off-by: Amine Hilaly <hilalyamine@gmail.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prow/jobs/jinja/presubmits/service_tests.jinja2 [hunk 1]
```diff
@@ -123,4 +123,25 @@
...
+    spec:
+      serviceAccountName: pre-submit-service-account
+      containers:
+      - image: {{ image_context.images["unit-test"] }}
+        resources:
+          limits:
+            cpu: 1
+          requests:
+            cpu: 1
+            memory: "256Mi"
```
Comment (RedbackThomson) [inline prow/jobs/jinja/presubmits/service_tests.jinja2:141]: I don't think we need an entire vCPU to run a shell script lol

File: prow/jobs/jinja/presubmits/service_tests.jinja2 [hunk 2]
```diff
@@ -123,4 +123,25 @@
...
 
+  - name: {{ service }}-metadata-file-test
+    decorate: true
+    optional: false
+    always_run: true
+    annotations:
+      karpenter.sh/do-not-evict: "true"
+    spec:
+      serviceAccountName: pre-submit-service-account
+      containers:
```
Comment (RedbackThomson) [inline prow/jobs/jinja/presubmits/service_tests.jinja2:134]: You need to include the `test-infra` repository as an extra ref, and set it as the default cwd, to use the new script

**Review by RedbackThomson — CHANGES_REQUESTED** [review]

File: Makefile [hunk 3]
```diff
@@ -31,6 +31,9 @@ kind-helm-test: ## Run the Helm tests for SERVICE
 test-recommended-policy:
 	@AWS_SERVICE=$(AWS_SERVICE) source ./scripts/iam-policy-test-runner.sh && assert_iam_policies
 
+test-metadata-file:
+	@AWS_SERVICE=$(AWS_SERVICE) source ./scripts/metadata-file-test-runner.sh && assert_metadata_file
```
[the remark this answers is not in the document]
Comment (A-Hilaly) [reply Makefile:35]: Good catch, deferring to the makefile to keep consistency with `test-recommended-policy`

File: prow/jobs/jinja/presubmits/service_tests.jinja2 (hunk 2 above)
Comment (A-Hilaly) [reply prow/jobs/jinja/presubmits/service_tests.jinja2:139]: Good catch. Added

**Review by RedbackThomson — APPROVED** [review]
Yay 
/lgtm

**Pull request merged**